doc/compiling/docker.md

```diff
@@ -0,0 +1,25 @@
+# Compiling minetestserver with Docker

```

Comment:
I would rather name this "Developing minetestserver with Docker" since it doesn't deal with creating or using the final end-user image at all.